src/test/java/benchmark/NQBenchmark2.java

```diff
@@ -62,15 +71,48 @@ private String readFromClasspath(String file) throws IOException {
 
     @Benchmark
     @Warmup(iterations = 2)
-    @Measurement(iterations = 100, time = 10)
+    @Measurement(iterations = 5, time = 10)
     @Threads(1)
     @Fork(3)
     @BenchmarkMode(Mode.AverageTime)
     @OutputTimeUnit(TimeUnit.MILLISECONDS)
     public NormalizedQueryTree benchMarkAvgTime(MyState myState) throws ExecutionException, InterruptedException {
         NormalizedQueryTree normalizedQuery = NormalizedQueryTreeFactory.createNormalizedQuery(myState.schema, myState.document, null, Collections.emptyMap());
-        System.out.println("fields size:" + normalizedQuery.getFieldToNormalizedField().size());
+//        System.out.println("fields size:" + normalizedQuery.getFieldToNormalizedField().size());
         return normalizedQuery;
     }
 
+    public static void main(String[] args) {
+        MyState myState = new MyState();
+        myState.setup();
+        NormalizedQueryTree normalizedQuery = NormalizedQueryTreeFactory.createNormalizedQuery(myState.schema, myState.document, null, Collections.emptyMap());
+//        System.out.println(printTree(normalizedQuery));
+        ImmutableListMultimap<Field, NormalizedField> fieldToNormalizedField = normalizedQuery.getFieldToNormalizedField();
+        System.out.println(fieldToNormalizedField.size());
+//        for (Field field : fieldToNormalizedField.keySet()) {
+//            System.out.println("field" + field);
+//            System.out.println("nf count:" + fieldToNormalizedField.get(field).size());
+//            if (field.getName().equals("field49")) {
+//                ImmutableList<NormalizedField> normalizedFields = fieldToNormalizedField.get(field);
+//                for (NormalizedField nf : normalizedFields) {
+//                    System.out.println(nf);
+//                }
+//            }
+//        }
+//        System.out.println("fields size:" + normalizedQuery.getFieldToNormalizedField().size());
+    }
+
+    static List<String> printTree(NormalizedQueryTree queryExecutionTree) {
+        List<String> result = new ArrayList<>();
+        Traverser<NormalizedField> traverser = Traverser.depthFirst(NormalizedField::getChildren);
+        traverser.traverse(queryExecutionTree.getTopLevelFields(), new TraverserVisitorStub<NormalizedField>() {
+            @Override
+            public TraversalControl enter(TraverserContext<NormalizedField> context) {
+                NormalizedField queryExecutionField = context.thisNode();
+                result.add(queryExecutionField.printDetails());
+                return TraversalControl.CONTINUE;
+            }
+        });
+        return result;
```

Comment:
Should something like this go on NormalisedQuery itself??

Reply:
moved

Reply:
maybe yes ... but I will not do it now.